partitura/io/exportkern.py

```diff
@@ -0,0 +1,327 @@
+                            self.out_data[row_idx, staff_idx] = kern_el
+                elif isinstance(el, spt.Tempo):
+                    # Apply tempo to all splines
+                    kern_el = f"*MM{to_quarter_tempo(el.qpm)}"
+                    self.out_data[row_idx] = kern_el
+                elif isinstance(el, spt.Measure):
+                    # Apply measure to all splines
+                    kern_el = f"={el.number}"
+                    self.out_data[row_idx] = kern_el
+                elif isinstance(el, spt.TimeSignature):
+                    # Apply element to all splines
+                    kern_el = f"*M{el.beats}/{el.beat_type}"
+                    self.out_data[row_idx] = kern_el
+                elif isinstance(el, spt.KeySignature):
+                    # Apply element to all splines
+                    if el.fifths < 0:
+                        alters = "-".join(KEYS[: el.fifths])
+                    elif el.fifths > 0:
+                        alters = "#".join(KEYS[: el.fifths])
+                    else:
+                        alters = ""
+                    kern_el = f"*k[{alters}]"
+                    self.out_data[row_idx] = kern_el
+                else:
+                    add_row = False
+                    warnings.warn(f"Element {el} is not supported for kern export yet.")
+                if add_row:
+                    row_idx += 1
+        return self.out_data
+
+    def trim(self, data):
+        # if an entire row is filled with "." elements remove it.
+        out_data = data[~np.all(data == ".", axis=1)]
+        return out_data
+
+    def sym_dur_to_kern(self, symbolic_duration: dict) -> str:
+        kern_base = KERN_DURS[symbolic_duration["type"]]
+        dots = (
+            "." * symbolic_duration["dots"]
+            if "dots" in symbolic_duration.keys()
+            else ""
+        )
+        if "actual_notes" in symbolic_duration.keys() and "normal_notes":
+            kern_base = (
+                int(kern_base)
+                * symbolic_duration["actual_notes"]
+                / symbolic_duration["normal_notes"]
+            )
+            kern_base = str(kern_base)
+        return kern_base + dots
+
+    def duration_to_kern(self, element: spt.GenericNote) -> str:
+        if isinstance(element, spt.GraceNote):
+            if element.grace_type == "acciaccatura":
+                return "p"
+            else:
+                return "q"
+        else:
+            if "type" not in element.symbolic_duration.keys():
+                warnings.warn(f"Element {element} has no symbolic duration type")
+                return "4"
+            return self.sym_dur_to_kern(element.symbolic_duration)
+
+    def pitch_to_kern(self, element: spt.GenericNote) -> str:
+        # To encode pitch correctly in kern we need to take into account the octave
+        # duplication of the step in kern can either move the note up or down an octave
```

Comment:
# To encode pitch correctly in kern we need to take into account that the octave
        # duplication of the step in kern can either move the note up or down an octave